### DISCUSSION TITLE
How to add features

### DISCUSSION THREAD

**Discussion by Tobe2d** [Kilvoctu/aiyabot#147] [Ideas]
I would like to add upscale to the list below the generated image

Already we have:
🖋 - edit prompt, then generate a new image with same parameters.
🎲 - randomize seed, then generate a new image with same parameters.
📋 - view the generated image's information.
❌ - deletes the generated image.

How to add upscale button here?

**Comment by Kilvoctu** [comment]
Heya, are you asking how to add that button on a fork of the aiyabot, or you're asking if it's possible to do so as-is right now?

If it's the former, to add the button, you can add it to the `DrawView(View)` class in viewhandler.py. You can use the existing buttons as a reference. Then the code for that button can run the `upscale_dream` code in upscalecog.py. You can reference the `quick_upscale` function in ctxmenuhandler.py to see how I do this for the context menu feature. If you have more questions, I can try answering any of them.

If it's the latter, then the aiyabot currently has no feature to add more buttons to the output. I've thought about adding additional buttons before, but I was undecided on what button can use the fifth slot on the row.

Also, I'll be moving this to Discussions. I'd prefer Issues to be only for bug reports.

**Reply by Tobe2d** [reply]
Actually I meant to ask for adding it ;-)
I tried adding it myself and I can see the button but I got lost in the script as I am not really a python developer but it was worth trying ;-)

![image](https://user-images.githubusercontent.com/4099839/225106091-bf4b6176-2d3c-4e21-82b3-7ef3c0c7e218.png)

The button is there but it is not functinal at all 😂

If you have time please consider adding it 🙏

**Reply by Kilvoctu** [reply]
Sorry, I don't know how I missed this reply. I try to reply to everyone within a day if I have something to say.
I do plan on adding some more buttons, it's just sorta low on my priorities list. I'll add it to the list of [projects](https://github.com/Kilvoctu/aiyabot/projects?query=is%3Aopen).

With the buttons, there can only be 5 per row, and I don't like clutter so there will be maximum 1 row. So the conundrum was that I wasn't sure what function would be worthy of the 5th slot on the row. What I was planning was something similar to this lovely [AIYA fork](https://github.com/killfrenzy96/AiyaFrenzyBot#modifications-from-aiyabot). They have a button that toggles the view to show more buttons.

**Reply by Tobe2d** [reply]
Hey, yes that would be amazing if we can get a single button that toggles the view to show more buttons.
Maybe 4 main buttons with most used ones and the 5th can be the one open up extra options.
